gengokit/httptransport/templates/client.go

```diff
@@ -217,15 +238,15 @@ func contextValuesToHttpHeaders(keys []string) httptransport.RequestFunc {
 	// error and attempt to decode the specific error message from the response
 	// body. Primarily useful in a client.
 	func DecodeHTTP{{$method.Name}}Response(_ context.Context, r *http.Response) (interface{}, error) {
+		defer r.Body.Close()
```

Comment:
@lelandbatey Am I missing something or were we really not doing this??

Reply:
We aren't doing this. I think the reason we're not doing this is because of [this line from the documentation on `http.Request.Body`:](https://golang.org/pkg/net/http/#Request.Body)

```
...
The Server will close the request body. The ServeHTTP
Handler does not need to.
```
I'm too under the weather at this moment to say for sure from my reading of that comment whether that means we definitely do not need to close `request.Body` or not, but I don't think adding these `defer r.Body.Close()` statements will hurt anything.